src/StateBridge.sol

```diff
@@ -163,14 +194,32 @@ contract StateBridge is FxBaseRootTunnel, Ownable {
         // This encoding is specified as the encoding of the `bytes` received by
         // `_processMessageFromRoot` in the Polygon state bridge. Specifically, it requires an ABI-
         // encoded tuple of `(uint256 newRoot, uint128 supersedeTimestamp)`.
-        message = abi.encode(root, timestamp);
+        message = abi.encodeWithSignature("receiveRoot(bytes)", abi.encode(root, timestamp));
```

Comment:
I would precompute this signature into a selector in the constructor. You also don't need to `abi.encode(...)` the arguments. It does that for you if you give `encodeWith{Selector, Signature}` the arguments.

Reply:
good catch, I need to update state bridge, forgot to do that with all the haste

Reply:
I don't think it makes sense to precompute anything here

Reply:
I did have an error in the function signature though, fixed that one